File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {
       if (nums.length == 0) return 0
       let map = new Map()
       for(let i=0; i< nums.length; i++)
       {
          if(map[nums[i]])
          {
            continue
          }
          else {
            map[nums[i]] = []
          }
        
       }

       for(let key of Object.keys(map))
       {
            if(map[key - 1])
            {
                map[key - 1].push(key)
            }
       }

       console.log(map)

       let maxLength = 1
       for(let key of Object.keys(map))
       {
            if(map[key].length > 0)
            {
                let m = map[key][0], j = 1
                while(map[m] != undefined)
                {
                    j++
                    m = map[m][0]
                }
                maxLength = Math.max(j, maxLength)
            }
       } 
       return maxLength
    }
}

File: Data Structures & Algorithms/is-anagram/submission-2.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        
        map = {}
        for i in list(s):
            if i in map:
                map[i] = map[i]+1
            else:
                map[i] = 1

        for i in list(t):
            if i not in map:
                return False

            map[i] = map[i]-1
            if map[i] < 0:
                return False
        
        return True
            
        